plugins/home-react/src/alpha/blueprints/CustomHomepageWidgetBlueprint.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { lazy, ReactElement } from 'react';
+import { compatWrapper } from '@backstage/core-compat-api';
+import { attachComponentData } from '@backstage/core-plugin-api';
+import {
+  coreExtensionData,
+  createExtensionBlueprint,
+  ExtensionBoundary,
+} from '@backstage/frontend-plugin-api';
+import {
+  CardConfig,
+  CardExtension,
+  CardExtensionProps,
+  CardLayout,
+  CardSettings,
+  ComponentParts,
+} from '../../extensions';
+
+/** @alpha */
+export interface CustomHomepageWidgetBlueprintParams {
+  /**
+   * Unique name for the widget. This value is exposed through component data
+   * and must match when referencing widgets from layout configuration.
+   */
+  name: string;
```

Comment:
Names are typically optional.

```suggestion
  name?: string;
```

Implementers are free to leave out the name if they have just one of a kind, or there's a very decidedly "default" one among the things it provides. As an example - the catalog plugin uses the PageBlueprint to make a default landing page for the overview page that does NOT have a name, and that gets the ID `page:catalog`. Then it has the individual entity pages, and those also use the PageBlueprint and a more complex path, and the name `entity` such that its full ID becomes `page:catalog/entity`